File: python/test_solution.py
import random
import unittest
from solution import solution

class TestSolution(unittest.TestCase):

    def test_solution_1000(self):
        self.assertEqual(233168,solution(1000))

    def test_solution_0(self):
        self.assertEqual(0,solution(0))

    def test_solution_10(self):
        self.assertEqual(23,solution(10))

if __name__ == '__main__':
    unittest.main()

File: ruby/test_solution.rb
require "./solution"
require "minitest/autorun"

# http://docs.seattlerb.org/minitest/Minitest/Assertions.html#method-i-assert

class TestSolution < Minitest::Test

  def test_fail
    assert(true, "Assertion was true")
  end

  def test_solution_limit_0
    assert_equal(0,solution(0))
  end

  def test_solution_limit_10
    assert_equal(23,solution(10))
  end

  def test_solution
    assert_equal(233168, solution(1000), "Solution is Correct")
  end

end